logistic_requisition_department/model/logistic_requisition.py

```diff
@@ -48,6 +49,12 @@ class LogisticRequisitionSource(models.Model):
     def _prepare_po_requisition(self, purch_req_lines, pricelist=None):
         res = super(LogisticRequisitionSource, self)._prepare_po_requisition(
             purch_req_lines, pricelist)
-
-        res['department_id'] = self.department_id.id
+        for line in self:
+            if line.department_id:
+                if 'department_id' in res and res['department_id'] != line.department_id.id:
+                    raise exceptions.Warning(
+                        'Cannot generate Purchase Order: '
+                        'the sourcing lines are from '
+                        'different departments.')
+                res['department_id'] = line.department_id.id
```

Comment:
https://www.odoo.com/documentation/8.0/reference/orm.html#other-recordset-operations

Mapped should remove duplicates if it is mapping record set